File: javascript/2021/01-Sonar-Sweep/part2.js
const fs = require(`fs`);

const input = fs.readFileSync(`input.txt`, `utf8`).split(`\n`).map(n => parseInt(n));

let lastDepth = input[0] + input[1] + input[2];

let numIncreases = 0;
for (let i = 1; i < input.length-2; i++) {
	let thisDepth = input[i] + input[i+1] + input[i+2];
	if (thisDepth > lastDepth) {
		numIncreases++;
	}
	
	lastDepth = thisDepth;
}

console.log(numIncreases);

File: python3/2021/01-Sonar-Sweep/part2.py
input = open('input.txt', 'r').read().split("\n")

depths = list(map(lambda s: int(s), input))

increases = 0;
for i in range(1, len(depths)):
	if depths[i-2] + depths[i-1] + depths[i] > depths[i-3] + depths[i-2] + depths[i-1]:
		increases += 1

print(increases)